game-core/src/main/java/games/strategy/engine/framework/map/download/MapDownloadProgressListener.java

```diff
@@ -17,8 +17,8 @@
   private static final int MAX_PROGRESS_VALUE = 100;
 
   private final DownloadFileDescription download;
-  private volatile Optional<Long> downloadLength = Optional.empty();
   private final JProgressBar progressBar;
+  @Nullable private Long downloadLength;
```

Comment:
Note that, with the above change, instances of this type are no longer thread-safe, as stated in the class Javadocs.

`downloadLength` is written by one thread and read by another.  While reading/writing an object reference is atomic in the sense that it guarantees no partial read/write from/to memory, changes to the reference made by one thread are not guaranteed to be visible to other threads without some kind of synchronization (so-called "safe publication").  Please see sections 3.4 and 3.5 of _Java Concurrency in Practice_ (specifically, section 3.4.2: _Example: Using Volatile to Publish Immutable Objects_, applies here).

Reply:
I agree, we have a mutable reference to an immutable object - added volatile back in: https://github.com/triplea-game/triplea/pull/5617